### PR TITLE
OCPBUGS-18480: Sync DaemonSet if operand image changes

### PR DESCRIPTION
**Pull request by jmencak**
NTO reconcilation loop assumed that it is only necessary to update operand's DaemonSet only when the RELEASE_VERSION changes.  However, during upgrades, with the introduction of multi-arch OCP, RELEASE_VERSION doesn't necessarily changes, whereas the operator and operand image can.

This change adds a condition to also update operand's DaemonSet on operand image changes.

Resolves: OCPBUGS-18480

### CODE REVIEW

**Review by djslavens — APPROVED** [review]

**Comment by jmencak** [comment]
/retest
@dagrayvid , if you have a minute, PTAL. Thank you.

**Comment by jmencak** [comment]
I feel this deserves
/label acknowledge-critical-fixes-only
otherwise multi-arch upgrades with the same release version are broken.

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -584,12 +586,24 @@ func (c *Controller) syncDaemonSet(tuned *tunedv1.Tuned) error {
...
-		// Update the DaemonSet
 		klog.V(2).Infof("syncDaemonSet(): operatorReleaseVersion (%s) != operandReleaseVersion (%s), updating", operatorReleaseVersion, operandReleaseVersion)
+		update = true
+	}
+
+	// OCPBUGS-18480: sync the DaemonSet also when the operand image changes
+	operandImageCurrent := ds.Spec.Template.Spec.Containers[0].Image
+	operandImageWanted := os.Getenv("CLUSTER_NODE_TUNED_IMAGE")
+	if operandImageCurrent != operandImageWanted {
+		klog.V(2).Infof("syncDaemonSet(): operandImageCurrent (%s) != operandImageWanted (%s), updating", operandImageCurrent, operandImageWanted)
```
Comment (Prashanth684) [inline pkg/operator/controller.go:598]: just curious - can this not be controlled by defining the daemon set deployment manifest and specifying the image in the image-references file?

Comment (jmencak) [reply pkg/operator/controller.go:598]: Not really.  Then we'd pass the responsibility for creating the DaemonSet to CVO and you'd not be able to handle operator's ManagementState.

**Pull request merged**